### PR TITLE
Add slint language server for .slint files

### PR DESCRIPTION
**Pull request by hunger**
https://slint-ui.com/ has the details to the language and LS implementation.

The binary is called slint_lsp (which is also the name in nvim-lspconfig), so I would like to keep that here as well.

### CODE REVIEW

**Review by williamboman — COMMENTED** [review]
Cool!! I'm however getting the following error when trying to run the binary on an Ubuntu 20.04 machine (x64):

```
./slint-lsp: error while loading shared libraries: libQt5Core.so.5: cannot open shared object file: No such file or directory
```

File: lua/nvim-lsp-installer/servers/slint_lsp/init.lua [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+                    return std.unzip_remote(ctx.github_release_file)
+                else
+                    return std.untargz_remote(ctx.github_release_file)
+                end
+            end),
+        },
+        default_options = {
+            cmd_env = {
+                PATH = process.extend_path { root_dir, "slint-lsp" },
+            },
```
Comment (williamboman) [inline lua/nvim-lsp-installer/servers/slint_lsp/init.lua:33]: By extending the PATH environment used when starting the server, we can avoid having to override the `cmd` set by lspconfig.

**Review by williamboman — APPROVED** [review]
Great work, LGTM!

**Comment by hunger** [comment]
> Cool!! I'm however getting the following error when trying to run the binary on an Ubuntu 20.04 machine (x64):
> 
> ```
> ./slint-lsp: error while loading shared libraries: libQt5Core.so.5: cannot open shared object file: No such file or directory
> ```

Those libraries are taken from the system and not part of the linux tarball. So I guess you just do not have those installed?

File: lua/nvim-lsp-installer/servers/slint_lsp/init.lua (hunk 1 above)
Comment (hunger) [reply lua/nvim-lsp-installer/servers/slint_lsp/init.lua:33]: Hey, thanks for fixing this!

**Comment by hunger** [comment]
One sec, give a a chance to test:-)

**Comment by hunger** [comment]
Ok, works for me again, now using PATH! Thanks for the patch putting me on the right track!

**Pull request merged**